src/software/networking/proto_multicast_sender.h

```diff
@@ -0,0 +1,44 @@
+#pragma once
+
+#include <boost/asio.hpp>
+#include <boost/bind.hpp>
+#include <string>
+
+template <class SendProto>
+class ProtoMulticastSender
+{
+   public:
+    /**
+     * Creates a ProtoMulticastSender that sends the SendProto over the network on the
+     * given address and port.
+     *
+     * @param io_service The io_service to use to service outgoing SendProto data
+     * @param ip_address The ip address of the multicast group to send data on
```

Comment:
javadoc should also document that the interface needs to be specified. Some examples here would also be super useful (since people will generally be less familiar with what providing an interface looks like)

Edit: also might want to change the parameter name too

Reply:
added examples here

the interface doesn't explicitly need to be specified, its just a property of the IPv6 addressing scheme. For IPv4 you _can't_ specify the interface, 

From Wikipedia:

> In order to resolve the ambiguity in textual addresses, a zone index must be appended to the address, the two separated by a percent sign (%).[10] The syntax of zone indices is an implementation-dependent string, although numeric zone indices must be universally supported as well. The link-local address
> fe80::1ff:fe23:4567:890a
> could be expressed by
> fe80::1ff:fe23:4567:890a%eth2
> 

emphasis on the "could be expressed" it will still work without

Reply:
didn't now that. Looks better now :+1: